src/common/http-signature/verify-signature.middleware.ts

```diff
@@ -1,30 +1,37 @@
 import { Injectable, NestMiddleware } from '@nestjs/common';
 import { NextFunction, Request, Response } from 'express';
-import { LTO } from '@ltonetwork/lto';
+import { LTO, getNetwork } from '@ltonetwork/lto';
 import { verify } from '@ltonetwork/http-message-signatures';
 
+export const lto = new LTO();
 @Injectable()
 export class VerifySignatureMiddleware implements NestMiddleware {
-  private readonly lto: LTO;
+  private lto: LTO;
 
   constructor() {
     this.lto = new LTO();
   }
 
-  async verify(req: Request, res: Response): Promise<boolean> {
+  async verifyRequest(req: Request, res: Response): Promise<boolean> {
     try {
+      const path = req.path;
+      const walletAddress = path.match(/3[^\/]*/)?.[0];
+      const network = getNetwork(walletAddress);
+
+      //switch to testnet if address is testnet
+      if (network == 'T') this.lto = new LTO(network);
```

Comment:
This never switches back to mainnet. No need to switch though. Just create a new LTO object for each request.

```typescript
const lto = new LTO(network)
```